service/storage_service.cc

```diff
@@ -2151,20 +2162,16 @@ future<> storage_service::raft_replace(raft::server& raft_server, raft::server_i
 
     // add myself to topology with request to replace
     while (!_topology_state_machine._topology.contains(raft_server.id())) {
-        slogger.info("raft topology: adding myself to topology for replace: {} replacing {}, ignored nodes: {}", raft_server.id(), replaced_id, ignored_ids);
         auto guard = co_await _group0->client().start_operation(&_abort_source);
 
         auto it = _topology_state_machine._topology.normal_nodes.find(replaced_id);
         if (it == _topology_state_machine._topology.normal_nodes.end()) {
             throw std::runtime_error(::format("Cannot replace node {}/{} because it is not in the 'normal' state", replaced_ip, replaced_id));
         }
 
-        for (const auto& ignored_id : ignored_ids) {
-            if (!_topology_state_machine._topology.find(ignored_id)) {
-                throw std::runtime_error(::format("ignored node {} is not found in the cluster", ignored_id));
-            }
-        }
+        auto ignored_ids = find_raft_nodes_from_hoeps(ignore_nodes_params);
```

Comment:
I understand that you moved the construction of `ignored_ids` here because now you're accessing `_topology_state_machine._topology` inside `find_raft_nodes_from_hoeps`, so you want to do it after `start_operation`.

This is something that would have been useful to get an explanation in the commit message (I had to scratch my head for a while trying to understand why you did this).

Anyway I think I'd prefer to keep the `_topology_state_machine._topology.find` check outside `find_raft_nodes_from_hoeps` -- the error message was previously mentioning "ignored node", now it's just generic "node {} not found" (and the user may wonder what it is about).

But it's probably not important, we can leave it as it is now. (And there is less code duplication this way.)